fluent-react/src/localization.ts

```diff
@@ -73,6 +79,194 @@ export class ReactLocalization {
+          this.reportError(error);
+        }
+        return createElement(Fragment, null, value);
+      }
+
+      return createElement(Fragment, null, componentToRender);
+    }
+
+    let localizedProps: Record<string, string> | undefined;
+    // The default is to forbid all message attributes. If the attrs prop exists
+    // on the Localized instance, only set message attributes which have been
+    // explicitly allowed by the developer.
+    if (args?.attrs && msg.attributes) {
+      localizedProps = {};
+      errors = [];
+      for (const [name, allowed] of Object.entries(args?.attrs)) {
+        if (allowed && name in msg.attributes) {
+          localizedProps[name] = bundle.formatPattern(
+            msg.attributes[name],
+            args?.vars,
+            errors
+          );
+        }
+      }
+      for (let error of errors) {
+        this.reportError(error);
+      }
+    }
+
+    // If the component to render is a known void element, explicitly dismiss the
+    // message value and do not pass it to cloneElement in order to avoid the
+    // "void element tags must neither have `children` nor use
+    // `dangerouslySetInnerHTML`" error.
+    if (typeof componentToRender.type === "string" && componentToRender.type in voidElementTags) {
+      return cloneElement(componentToRender, localizedProps);
+    }
+
+    // If the message has a null value, we're only interested in its attributes.
+    // Do not pass the null value to cloneElement as it would nuke all children
+    // of the wrapped component.
+    if (msg.value === null) {
+      return cloneElement(componentToRender, localizedProps);
+    }
+
+    errors = [];
+    const messageValue = bundle.formatPattern(msg.value, args?.vars, errors);
+    for (let error of errors) {
+      this.reportError(error);
+    }
+
+    // If the message value doesn't contain any markup nor any HTML entities,
+    // insert it as the only child of the component to render.
+    if (!reMarkup.test(messageValue) || this.parseMarkup === null) {
+      return cloneElement(componentToRender, localizedProps, messageValue);
+    }
+
+    let elemsLower: Map<string, ReactElement>;
+    if (args?.elems) {
+      elemsLower = new Map();
+      for (let [name, elem] of Object.entries(args?.elems)) {
+        elemsLower.set(name.toLowerCase(), elem);
+      }
+    }
+
+    // If the message contains markup, parse it and try to match the children
+    // found in the translation with the args passed to this function.
+    const translationNodes = this.parseMarkup(messageValue);
+    const translatedChildren = translationNodes.map(({ nodeName, textContent }) => {
+      if (nodeName === "#text") {
+        return textContent;
+      }
+
+      const childName = nodeName.toLowerCase();
+
+      // If the child is not expected just take its textContent.
+      if (
+        !elemsLower ||
+        !Object.prototype.hasOwnProperty.call(elemsLower, childName)
+      ) {
```

Comment:
```suggestion
      if (!elemsLower?.has(childName)) {
```

Reply:
Oh heh, was just typing a reply above that I'd forgotten to run the tests after the change to `Map`. Fix in 515918e7c701859f98930b1df2de61fc211cb866.